src/clients/bridges/ZKSyncAdapter.ts

```diff
@@ -162,6 +129,67 @@ export class ZKSyncAdapter extends BaseAdapter {
     return this.computeOutstandingCrossChainTransfers(l1Tokens);
   }
 
+  async queryL1BridgeInitiationEvents(
+    l1Token: string,
+    sender: string | null = null,
+    recipient: string | null = null,
+    searchConfig: EventSearchConfig
+  ): Promise<Event[]> {
+    const isWeth = this.isWeth(l1Token);
+    const isL2Contract = await this.isL2ChainContract(recipient);
+
+    let bridge: Contract;
+    let filter: EventFilter;
+    if (isWeth) {
+      // If sending WETH from EOA, we can assume the EOA is unwrapping ETH and sending it through the AtomicDepositor.
+      // If sending WETH from a contract, then the only event we can track from a ZkSync contract is the
+      // NewPriorityRequest event which doesn't have any parameters about the 'to' or 'amount' sent. Therefore, we must
+      // track the HubPool and assume any transfers we are tracking from contracts are being sent by the HubPool.
+      const hubPool = this.getHubPool();
+      const atomicDepositor = this.getAtomicDepositor();
+
+      [bridge, filter] = isL2Contract
+        ? [hubPool, hubPool.filters.TokensRelayed()]
+        : [atomicDepositor, atomicDepositor.filters.ZkSyncEthDepositInitiated(sender, recipient)];
+    } else {
+      bridge = this.getL1ERC20BridgeContract();
+      filter = bridge.filters.DepositInitiated(null, sender, recipient);
+    }
+
+    return paginatedEventQuery(bridge, filter, searchConfig);
+  }
+
+  getAddressAlias(l1Address: string): string {
+    return ethersUtils.getAddress(zksync.utils.applyL1ToL2Alias(l1Address));
+  }
+
+  async queryL2BridgeFinalizationEvents(
+    l1Token: string,
+    sender: string | null = null,
+    recipient: string | null = null,
+    searchConfig: EventSearchConfig
+  ): Promise<Event[]> {
+    let bridge: Contract;
+    let filter: EventFilter;
+
+    if (this.isWeth(l1Token)) {
+      if (sender === null && recipient !== null) {
+        sender = this.getAddressAlias(
+          (await this.isL2ChainContract(recipient)) ? this.getHubPool().address : this.getAtomicDepositor().address
+        );
+      }
+
+      bridge = this.getL2Eth();
+      filter = bridge.filters.Transfer(null, recipient);
```

Comment:
Should we have this?

```suggestion
      filter = bridge.filters.Transfer(sender, recipient);
```